examples/generate_data/generate_data.cc

```diff
@@ -24,137 +25,148 @@ Generate a binary file containing a fixed set of messages.
     return 1;
   }
 
+  // Enforce a 4-byte aligned address.
+  uint8_t storage[4096];
+  size_t alignment_offset = reinterpret_cast<size_t>(storage) % 4;
+  char* buffer = reinterpret_cast<char*>(storage + alignment_offset);
+
   //////////////////////////////////////////////////////////////////////////////
   // Write a pose message.
   //////////////////////////////////////////////////////////////////////////////
 
-  MessageHeader header;
-  header.crc = 0x40C57DE6;
-  header.message_type = MessageType::POSE;
-  header.payload_size_bytes = sizeof(PoseMessage);
+  MessageHeader* header = reinterpret_cast<MessageHeader*>(buffer);
+  *header = MessageHeader();
+  header->sequence_number = 0;
+  header->message_type = MessageType::POSE;
+  header->payload_size_bytes = sizeof(PoseMessage);
 
-  stream.write(reinterpret_cast<char*>(&header), sizeof(header));
+  PoseMessage* pose_message = reinterpret_cast<PoseMessage*>(header + 1);
```

Comment:
while clever, I think this is a bit unclear. I'd rather us just do the more straightforward buffer + sizeof(header)

Reply:
I was previously doing something like `buffer += sizeof(MessageHeader)` but it got a bit messy with the buffer alignment since I needed to keep track of 3 pointers: storage, start of aligned storage (for CRC calc and file write), current buffer offset. `alignas()` makes it cleaner so I can go back to the more clear way.